app/src/main/java/com/mapbox/maps/testapp/examples/snapshotter/MapSnapshotterActivity.kt

```diff
@@ -30,42 +30,39 @@ class MapSnapshotterActivity : AppCompatActivity(), Snapshotter.SnapshotReadyCal
       .pixelRatio(1.0f)
       .build()
 
-    mapSnapshotter = Snapshotter(this, snapshotterOptions)
-    mapSnapshotter.setUri(Style.MAPBOX_STREETS)
-    mapSnapshotter.setCameraOptions(
-      CameraOptions.Builder().zoom(14.0).center(
-        Point.fromLngLat(
-          4.895033, 52.374724
-        )
-      ).build()
-    )
-    mapSnapshotter.start(this)
-  }
-
-  override fun onStyleLoaded(style: Style) {
-    Logger.e(TAG, "OnStyleLoaded: ${style.styleURI}")
-  }
-
-  override fun onSnapshotCreated(snapshot: Expected<MapSnapshotInterface?, String?>) {
-    if (snapshot.isValue) {
-      val mapSnapshot = snapshot.value as MapSnapshotInterface
-      Logger.e(TAG, "Result: ${mapSnapshot.image().data.size}")
+    mapSnapshotter = Snapshotter(this, snapshotterOptions).apply {
+      setStyleListener(this@MapSnapshotterActivity)
+      setUri(Style.MAPBOX_STREETS)
+      setCameraOptions(
+        CameraOptions.Builder().zoom(14.0).center(
+          Point.fromLngLat(
+            4.895033, 52.374724
+          )
+        ).build()
+      )
+      start {
+        it?.let { mapSnapshot ->
+          Logger.i(TAG, "Result: ${mapSnapshot.image().data.size}")
```

Comment:
nit: log can be removed

Reply:
I did not touch that logic, I guess author of that example did it intentionally.